test/e2e/annotations/canary.go

```diff
@@ -475,6 +475,85 @@ var _ = framework.DescribeAnnotation("canary-*", func() {
 		})
 	})
 
+	ginkgo.Context("when canaried by header with value and use regex to match", func() {
+		ginkgo.It("should route requests to the correct upstream", func() {
```

Comment:
Instead of `Context`, I suggest to just use

> ginkgo.It("routes to canary when head-value is a pattern and match-strategy is regex")

Reply:
done.